### DISCUSSION TITLE
History and number of iterations?

### DISCUSSION THREAD

**Discussion by cherifM** [mdolab/pyoptsparse#329] [General]
Hi , 
im using  pyoptspare as my main optimizer, im facing issue with optview (i dont x11 because im run it in server), so i did use python and im trying to view my opitmization progress, if I use SLSQP,  i get :
```
  ---------------------------------------------------------------------------
     START OF THE SEQUENTIAL LEAST SQUARES PROGRAMMING ALGORIT
   ---------------------------------------------------------------------------

     PARAMETERS:
        ACC =   0.1000D-08
        MAXITER = 10
        IPRINT =   1
IOUT =   6


     ITER =    1     OBJ = -0.10001259E+01
     ITER =    2     OBJ = -0.10001105E+01
     ITER =    3     OBJ = -0.10002035E+01
     ITER =    4     OBJ = -0.10002023E+01
     ITER =    5     OBJ = -0.10002176E+01
     ITER =    6     OBJ = -0.10001288E+01
     ITER =    7     OBJ = -0.10004467E+01
     ITER =    8     OBJ = -0.10004755E+01
     ITER =    9     OBJ = -0.10004219E+01
     ITER =   10     OBJ = -0.10007070E+01
        NUMBER OF FUNC-CALLS:  NFUNC = 103
        NUMBER OF GRAD-CALLS:  NGRAD =  11
       
```
now , in my python code,  i did :

```
import  pyoptsparse.pyOpt_history  as History 
hist = History.History("hist.hist" , flag="r")
last=  ( hist.callCounters[-1]  )
last 
'113'
```

why im getting 113 , should be not 10 ?  I just want to plot progress each iteration, any help ? 

thanks in advance

cherif

**Comment by sseraj** [comment]
`callCounters` includes every function and gradient call. In your case, you have 10 major iterations, but 114 total function and gradient calls.

Depending on how frequently you want to view the optimization progress, it might be easier to transfer the history file to your local machine and use OptView. Even if you get the plotting working on the server, you will need something like X11 to see the plot.
